pkg/router/gloo.go

```diff
@@ -288,7 +289,37 @@ func (gr *GlooRouter) createFlaggerUpstream(canary *flaggerv1.Canary, upstreamNa
 		}
 	} else if err != nil {
 		return fmt.Errorf("upstream %s.%s get query error: %w", upstreamName, canary.Namespace, err)
+	} else {
+		return gr.syncUpstreamSpec(curUpstream, canary)
+	}
+	return nil
+}
+
+func (gr *GlooRouter) syncUpstreamSpec(curUpstream *gloov1.Upstream, canary *flaggerv1.Canary) error {
+	glooUpstreamWithConfig, err := gr.getGlooConfigUpstream(canary)
+	if err != nil {
+		return err
+	}
+
+	if glooUpstreamWithConfig == nil {
+		return nil
 	}
+
+	glooUpstreamLB := glooUpstreamWithConfig.Spec.LoadBalancerConfig
+	loadBalancerDiff := cmp.Diff(glooUpstreamLB, curUpstream.Spec.LoadBalancerConfig)
+
+	if loadBalancerDiff != "" {
+		gr.logger.Debugf("detect diff in upstream spec %s.%s %s", curUpstream.Name, canary.Namespace, loadBalancerDiff)
+
+		cloneUpstream := curUpstream.DeepCopy()
+		cloneUpstream.Spec.LoadBalancerConfig = glooUpstreamLB
+
+		_, err = gr.glooClient.GlooV1().Upstreams(canary.Namespace).Update(context.TODO(), cloneUpstream, metav1.UpdateOptions{})
```

Comment:
FYI: You can try to update it using `patch`. This way you can avoid using deepcopy.

Reply:
What more advantages are of using Patch instead of Update in this particular usecase?

Reply:
I think the update is Spec.LoadBalancerConfig single field, using patch can reduce the use of deepcopy.